### PR TITLE
DAOS-5605 build: add lz4 dependency

### PR DESCRIPTION
**Pull request by johannlombardi**
With the introduction of online compression, we need to link with
the lz4 library now. Add it to the docker images.

Signed-off-by: Johann Lombardi <johann.lombardi@intel.com>

### CODE REVIEW

File: utils/docker/Dockerfile.centos.7 [hunk 1]
```diff
@@ -62,7 +62,7 @@ RUN yum -y install epel-release; \
         python-pep8 python-pygit2 python2-pygithub python-requests     \
         readline-devel scons sg3_utils ShellCheck yasm pciutils        \
         valgrind-devel python36-pylint man java-1.8.0-openjdk maven    \
-        json-c-devel python36-requests
+        json-c-devel python36-requests lz4 lz4-devel
```
Comment (brianjmurrell) [inline utils/docker/Dockerfile.centos.7:65]:
```suggestion
        json-c-devel python36-requests lz4-devel
```

`lz4-devel` will already requires the `lz4` library.  Letting it be implicit means we don't have to do anything if upstream changes packaging for any reason.

File: utils/docker/Dockerfile.leap.15 [hunk 2]
```diff
@@ -63,7 +63,7 @@ RUN zypper --non-interactive --no-gpg-checks refresh; \
            readline-devel sg3_utils which yasm                        \
            python-devel python3-devel valgrind-devel hwloc-devel      \
            openmpi3-devel man fuse3-devel patchelf scons lua-lmod     \
-           libjson-c-devel
+           libjson-c-devel liblz4 liblz4-devel
```
Comment (brianjmurrell) [inline utils/docker/Dockerfile.leap.15:66]: Ditto.

```suggestion
           libjson-c-devel liblz4-devel
```

File: utils/docker/Dockerfile.ubuntu.20.04 [hunk 3]
```diff
@@ -29,7 +29,8 @@ RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get install -y \
             locales make meson nasm ninja-build pandoc patch            \
             pylint python-dev python3-dev scons sg3-utils uuid-dev      \
             yasm valgrind libhwloc-dev man fuse3 libfuse3-dev           \
-            openjdk-8-jdk maven libopenmpi-dev patchelf libjson-c-dev        
+            openjdk-8-jdk maven libopenmpi-dev patchelf libjson-c-dev	\
+	    lz4 liblz4-dev
```
Comment (brianjmurrell) [inline utils/docker/Dockerfile.ubuntu.20.04:33]: Ditto.

```suggestion
	    liblz4-dev
```

**Review by brianjmurrell — CHANGES_REQUESTED** [review]
Seems like the `daos.spec` will also need a `BuildRequires: liblz4-devel` so that the RPMs are built with it.  Given that the RPM builds succeeded for this PR, it must still be an optional scons dependency, yes?

**Review by ryon-jensen — COMMENTED** [review]
Looks good to me, though I'm not sure I'm the best person to review dockerfile changes. Definitely address Brian's feedback :)

**Comment by johannlombardi** [comment]
> Given that the RPM builds succeeded for this PR, it must still be an optional scons dependency, yes?

The code to use lz4 hasn't been landed yet.

File: utils/rpms/daos.spec [hunk 4]
```diff
@@ -38,8 +38,10 @@ BuildRequires: libpmem-devel >= 1.8, libpmemobj-devel >= 1.8
 BuildRequires: fuse3-devel >= 3.4.2
 %if (0%{?suse_version} >= 1500)
 BuildRequires: libprotobuf-c-devel
+BuildRequires: liblz4-devel
 %else
 BuildRequires: protobuf-c-devel
+BuildRequires: lz4-devel
```
Comment (johannlombardi) [inline utils/rpms/daos.spec:44]: @brianjmurrell It sounds like the package has a different name on centos and leap. Did i insert this in the right place? I wasn't sure ...

Comment (brianjmurrell) [reply utils/rpms/daos.spec:44]: Seems good now at least.

**Comment by brianjmurrell** [comment]
> The code to use lz4 hasn't been landed yet.

Ahhh.  That would explain it then.  Should this PR be merged into the PR that will land the code to use lz4 then, just to keep the commit to support lz4 atomic?

**Comment by johannlombardi** [comment]
@brianjmurrell yeah, trying to make the packaging works for now. Is the leap 15 failure expected?

**Comment by brianjmurrell** [comment]
> @brianjmurrell yeah, trying to make the packaging works for now. Is the leap 15 failure expected?

It's not.  Perhaps a transient issue.  Might be worth restarting from the test stage just to confirm.

**Review by brianjmurrell — APPROVED** [review]

**Review by ryon-jensen — APPROVED** [review]

**Pull request merged**